### DISCUSSION TITLE
3.7 to 4.0 Jointjs Upgrade

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2565] [Q&A]
### Introduction

I have downloaded the latest version.I am trying to run demo apps in the provided apps folder.
npm install is fully stuck and not installing for any app.

I am able to install via npm i @joint/core succesfully
But I was trying to resolve one error related to directed graph via installing package
npm i @joint/layout-directed-graph
But it is giving error
``` npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://registry.npmjs.org/@joint%2flayout-directed-graph failed, reason: getaddrinfo ENOTFOUND registry.npmjs.org
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

npm ERR! A complete log of this run can be found in:

```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
Isn't this related to:
https://stackoverflow.com/questions/23193614/npm-err-network-getaddrinfo-enotfound
https://github.com/nodejs/help/issues/4093

**Reply by ROYopedia** [reply]
Let me take a look. But for all apps npm i is stuck for all. Getting no errors.

 
 Do you have any migration process to 4.0 guide available for rappid 3.7.0 users? Or the change log is sufficient?

**Reply by kumilingus** [reply]
Our [changelog](https://changelog.jointjs.com/plus/4.0.0/changelog#breaking-changes) contains migration guide.
